lib/CXGN/BrAPI/v2/ObservationUnits.pm

```diff
@@ -157,44 +172,70 @@ sub search {
 
         if ($geo_coordinates_string){
             $geo_coordinates = decode_json $geo_coordinates_string;
-        } 
+        }
+
+        my $additional_info;
+        my $additional_info_type_id = SGN::Model::Cvterm->get_cvterm_row($self->bcs_schema, 'stock_additional_info', 'stock_property')->cvterm_id();
+        my $rs = $self->bcs_schema->resultset("Stock::Stockprop")->search({ type_id => $additional_info_type_id, stock_id => $obs_unit->{observationunit_stock_id} });
+        if ($rs->count() > 0){
+            my $additional_info_json = $rs->first()->value();
+            $additional_info = $additional_info_json ? decode_json($additional_info_json) : undef;
+        }
+        # Add plot number and plant number to additional info
+        if ($obs_unit->{obsunit_plot_number}) {$additional_info->{'plotNumber'} = $obs_unit->{obsunit_plot_number}};
+        if ($obs_unit->{obsunit_plant_number}) {$additional_info->{'plantNumber'} = $obs_unit->{obsunit_plant_number}};
 
         my $entry_type = $obs_unit->{obsunit_is_a_control} ? 'check' : 'test';
 
         my $replicate = $obs_unit->{obsunit_rep};
         my $block = $obs_unit->{obsunit_block};
-        my $plot = $obs_unit->{obsunit_plot_number};
-        my $plant = $obs_unit->{obsunit_plant_number};
+        my $plot;
+        my $plant;
+        if ($obs_unit->{observationunit_type_name} eq 'plant') {
+            $plant = $obs_unit->{observationunit_stock_id};
+            if (%plant_parents{$obs_unit->{observationunit_stock_id}}) {
+                $plot = %plant_parents{$obs_unit->{observationunit_stock_id}};
+            }
+        } else {
+            $plot = $obs_unit->{observationunit_stock_id};
+        }
 
         my $level_name = $obs_unit->{observationunit_type_name};
         my $level_order = _order($level_name) + 0;
-        my $level_code = eval "\$$level_name" || "";
+        my $level_code = $obs_unit->{observationunit_stock_id};
```

Comment:
It won't be always the case. When observation units are plots, value should correspond to the plot number. It will depend on the level

Reply:
For this change, I am setting the level code to be the object id so that we can form parent-child relationships through the `observationLevelRelationships` field. I moved the `plotNumber` to an additional info field. Do you know of anything in brapi that the plotNumber matches to? I couldn't find anything.

Reply:
observationUnitDbId can be used to create the parent-child relationship, isn't it?.
Yeah, there is nothing specific for numbers. So, seems like levelCode fits better for plotNumber, plantNumber, etc.  within the observationUnitPosition field.

Reply:
You can use the `observationUnitDbId` to indicate the id of that object, but not of the parent or child object. So if you have an object like 

```
{
  "observationUnitName": "test plant", 
  "observationUnitDbId": "123", 
  "observationUnitPosition": {
    "observationLevel": {
      "levelName": "plant",
      "levelCode": "123"
    },
    "observationLevelRelationship": [
        {
           "levelName": "plot",
           "levelCode": "456"
        },
        {
           "levelName": "plant",
           "levelCode": "123"
        }
     ]
  }
```

You know that the parent is a plot with an id 456. And likewise when making a request for plot 456 you would get a response. It seems like it would be inconsistent to use the levelCode for the object id sometimes, but the plot number other times. 

I'm happy to have a call about this if you think it would be easier to talk it through together!

Reply:
And just to get a full picture, an object doesn't return its child relationships, only its parent relationships. So if you were to get plot with an id of 456 out you would get this response:

```
{
  "observationUnitName": "test plot", 
  "observationUnitDbId": "456", 
  "observationUnitPosition": {
    "observationLevel": {
      "levelName": "plot",
      "levelCode": "456"
    },
    "observationLevelRelationship": [
        {
           "levelName": "plot",
           "levelCode": "456"
        }
     ]
  }
```

I'm thinking we don't want to return the children because you can't trace a tree structure for the `observationLevelRelationship` single array structure. If you just had plant, that would be possible. But say you had tissue under plant as well. You would know all of the plants are children of the plot. But the tissue relationships might not be useful because you wouldn't know which tissue sample belonged to which plant. That might not be important though, maybe you just want to know all of the tissue samples in that plot. Anyway, I didn't return the child relationships, maybe something to add later if it is useful.

Reply:
Do you think you will be able to join our lab meeting tomorrow? We can discuss there

Reply:
Reverted back to plot number for level code